cfme/markers/env_markers/provider.py

```diff
@@ -81,6 +138,73 @@ def parametrize(metafunc, argnames, argvalues, *args, **kwargs):
             )(metafunc.function)
 
 
+class DataProvider(object):
```

Comment:
**optional** use `attr.s` and a `attr.Factory` with takes_self=True to set klass (also init=False)